starlette_web/common/ws/base_endpoint.py

```diff
@@ -2,78 +2,138 @@
 import json
 import logging
 from json import JSONDecodeError
-from typing import ClassVar, Type, Any
+from typing import ClassVar, Type, Any, Optional, List, Union, Dict, Tuple
+import sys
 
-from marshmallow import Schema
-from sqlalchemy.exc import InvalidRequestError
+import anyio
+from anyio._core._tasks import TaskGroup
+from marshmallow import Schema, ValidationError
 from starlette.endpoints import WebSocketEndpoint
-from starlette.websockets import WebSocket
-
-from common.utils.async_utils import create_task
-from contrib.auth.models import User
-from starlette_web.contrib.auth.backend import JWTAuthenticationBackend
+from starlette.types import Scope, Receive, Send
+from starlette.websockets import WebSocket, WebSocketDisconnect
+
+from starlette_web.common.authorization.backends import (
+    BaseAuthenticationBackend, NoAuthenticationBackend,
+)
+from starlette_web.common.authorization.base_user import BaseUserMixin
+from starlette_web.common.authorization.permissions import BasePermission, OperandHolder
+from starlette_web.common.http.exceptions import PermissionDeniedError
 from starlette_web.common.ws.requests import WSRequest
-from starlette_web.common.ws.schemas import WSRequestAuthSchema
-from starlette_web.common.authorization.backends import BaseAuthenticationBackend
+from starlette_web.common.utils.crypto import get_random_string
 
 
 logger = logging.getLogger(__name__)
 
 
 class BaseWSEndpoint(WebSocketEndpoint):
-    auth_backend: ClassVar[Type[BaseAuthenticationBackend]] = JWTAuthenticationBackend
-    request_schema: ClassVar[Type[Schema]] = WSRequestAuthSchema
-    user: User
+    auth_backend: ClassVar[Type[BaseAuthenticationBackend]] = NoAuthenticationBackend
+    permission_classes: ClassVar[List[Union[Type[BasePermission], OperandHolder]]] = []
```

Comment:
а у тебя py3.10+? если так, то как думаешь на счет такой вариации тайп-хинтинга?
```suggestion
    permission_classes: ClassVar[list[Type[BasePermission] | OperandHolder]] = []
```

Reply:
отложил до https://github.com/DmitryBurnaev/starlette-web/issues/25 )))
вообще, я себе поставил новый pycharm, но питон у меня пока 3.9.5

Reply:
ок) пусть будет пока с поддержкой 3.9) я не против))